node-test/src/test/scala/org/bitcoins/node/BroadcastTransactionTest.scala

```diff
@@ -0,0 +1,88 @@
+package org.bitcoins.node
+
+import org.bitcoins.testkit.wallet.BitcoinSWalletTest
+import org.scalatest.FutureOutcome
+import org.bitcoins.node.config.NodeAppConfig
+import org.bitcoins.chain.config.ChainAppConfig
+import org.bitcoins.node.models.Peer
+import org.bitcoins.chain.models.BlockHeaderDAO
+import org.bitcoins.chain.blockchain.ChainHandler
+import org.bitcoins.testkit.node.NodeTestUtil
+import org.bitcoins.core.currency._
+import org.bitcoins.core.wallet.fee.SatoshisPerByte
+import org.bitcoins.rpc.util.AsyncUtil
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.scalactic.Bool
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import org.bitcoins.testkit.async.TestAsyncUtil
+
+class BroadcastTransactionTest extends BitcoinSWalletTest {
+
+  override type FixtureParam = WalletWithBitcoind
+
+  def withFixture(test: OneArgAsyncTest): FutureOutcome =
+    withFundedWalletAndBitcoind(test)
+
+  it must "broadcast a transaction" in { param =>
+    val WalletWithBitcoind(wallet, rpc) = param
+
+    /**
+      * This is not ideal, how do we get one implicit value (`config`)
+      * to resolve to multiple implicit parameters?
+      */
+    implicit val nodeConfig: NodeAppConfig = config
+    implicit val chainConfig: ChainAppConfig = config
+
+    def hasSeenTx(transaction: Transaction): Future[Boolean] = {
+      rpc
+        .getRawTransaction(transaction.txIdBE)
+        .map { _ =>
+          true
+        }
+        .recover {
```

Comment:
nit: It actually might be nice to adjust the `getRawTransaction()` rpc call to incorporate this logic and return a `Future[Option[...]]` cc: @cwaldron97

Reply:
That's not how Core does it, though. If we're going to start applying that sort of logic there's _a lot_ of places that can happen. And if we're 100% consistent with Core some places but take liberties other places, we end up with a very confusing library.

Reply:
I don't know if it relevant but in V18 `getRawTransaction` fundamentally changes how it works and no longer check unspent `UTXO` for a transaction. 
`"The remaining behaviors are as follows: 1. If a blockhash is provided, check the corresponding block. 2. If no blockhash is provided, check the mempool. 3. If no blockhash is provided but txindex is enabled, also check txindex."`

Reply:
We're looking for transactions in the mempool (the mempool are the set of transactions a node has received, but not yet confirmed in a block), so I don't think this matters to us here.